crates/primitives/src/transaction/tx_type.rs

```diff
@@ -15,6 +15,10 @@ pub const EIP1559_TX_TYPE_ID: u8 = 2;
 /// Identifier for [TxEip4844](crate::TxEip4844) transaction.
 pub const EIP4844_TX_TYPE_ID: u8 = 3;
 
+/// Identifier for [TxDeposit](crate::TxDeposit) transaction.
+#[cfg(feature = "optimism")]
+use crate::DEPOSIT_TX_TYPE;
+
```

Comment:
I think this was the old convention - we should define the concrete type id here following the canonical pattern above. Will also have to delete the `crate::DEPOSIT_TX_TYPE` value and use this in place.
```suggestion
/// Identifier for [TxDeposit](crate::TxDeposit) transaction.
#[cfg(feature = "optimism")]
pub const crate::DEPOSIT_TX_TYPE_ID: u8 = 126;

```